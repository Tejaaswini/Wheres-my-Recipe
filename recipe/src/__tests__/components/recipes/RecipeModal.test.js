import _ from 'lodash';
import React from 'react';
import ReactDOM from 'react-dom';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import reduxThunk from 'redux-thunk';
import { connect } from 'react-redux';
import { Provider } from 'react-redux';
import { Link } from 'react-router-dom';
import {configure, mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import { BrowserRouter as Router } from 'react-router-dom';
import RecipeModal from '../../../components/recipes/recipe_modal';

const middlewares = [reduxThunk];
const mockStore = configureStore(middlewares);

configure({adapter: new Adapter()});

describe('Recipes Creation', () => {
    let store, wrapper;
    let state = {
        auth: {
            authenticated: true
        },
        categories: {},
        category: {},
        recipes: {},
        recipe: {},
    };

    beforeEach(() => {
        store = mockStore(state);
    });

    it('it renders without crashing', () => {
        shallow(
            <Provider store={store}>
                <Router>
                    <RecipeModal />
                </Router>
            </Provider>
        );
    });

    it('it has valid snapshot', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Router>
                    <RecipeModal />
                </Router>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});