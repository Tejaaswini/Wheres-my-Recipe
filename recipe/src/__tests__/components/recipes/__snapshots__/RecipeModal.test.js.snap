// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes Creation it has valid snapshot 1`] = `
<div>
  <div
    aria-hidden="true"
    aria-labelledby="recipeModalTitle"
    className="modal fade"
    id="recipeModal"
    role="dialog"
    tabindex="-1"
  >
    <div
      className="modal-dialog modal-dialog-centered"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title"
            id="recipeModalLongTitle"
          >
            Create New Recipe
          </h5>
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            id="close"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
        <div
          className="modal-body"
        >
          <div>
            <form
              className=""
              id="recipeForm"
              name="recipeForm"
              onSubmit={[Function]}
            >
              <div
                className="row form-group"
              >
                <label
                  className="col-md-3 col-form-label"
                  htmlFor="recipe_name"
                >
                  Recipe name
                </label>
                <div
                  className="col col-md-9"
                >
                  <small
                    className="form-text text-muted"
                  />
                  <input
                    className="form-control"
                    id="recipe_name"
                    name="name"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onFocus={[Function]}
                    type="text"
                    value=""
                  />
                  <div
                    className="invalid-feedback"
                  >
                    
                  </div>
                </div>
              </div>
              <div
                className="row form-group"
              >
                <label
                  className="col-md-3 col-form-label"
                  htmlFor="description"
                >
                  Description
                </label>
                <div
                  className="col col-md-9"
                >
                  <small
                    className="form-text text-muted"
                  />
                  <textarea
                    className="form-control"
                    id="description"
                    name="description"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onFocus={[Function]}
                    value=""
                  />
                  <div
                    className="invalid-feedback"
                  >
                    
                  </div>
                </div>
              </div>
              <div
                className="row form-group"
              >
                <label
                  className="col-md-3 col-form-label"
                  htmlFor="ingredients"
                >
                  Ingredients
                </label>
                <div
                  className="col col-md-9"
                >
                  <small
                    className="form-text text-muted"
                  >
                    Type with space and each on new line.
                  </small>
                  <textarea
                    className="form-control"
                    id="iingredients"
                    name="ingredients"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onFocus={[Function]}
                    value=""
                  />
                  <div
                    className="invalid-feedback"
                  >
                    
                  </div>
                </div>
              </div>
              <div
                className="row form-group"
              >
                <label
                  className="col-md-3 col-form-label"
                  htmlFor="directions"
                >
                  Directions
                </label>
                <div
                  className="col col-md-9"
                >
                  <small
                    className="form-text text-muted"
                  >
                    Type with space and each on new line.
                  </small>
                  <textarea
                    className="form-control"
                    id="directions"
                    name="directions"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onFocus={[Function]}
                    value=""
                  />
                  <div
                    className="invalid-feedback"
                  >
                    
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            type="button"
          >
            Cancel
          </button>
          <button
            className="btn btn-success"
            form="recipeForm"
            type="submit"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    id="notification-wrapper"
  />
</div>
`;
